problem-solver/cxx/inferenceModule/generator/SolutionTreeGenerator.cpp

```diff
@@ -59,19 +60,38 @@ bool SolutionTreeGenerator::addNode(ScAddr const & formula, ScTemplateParams con
   return result;
 }
 
-ScAddr SolutionTreeGenerator::createSolutionNode(ScAddr const & formula, ScTemplateParams const & templateParams)
+ScAddr SolutionTreeGenerator::createSolutionNode(ScAddr const & formula, ScTemplateParams const & templateParams, std::set<std::string> const & varNames)
 {
-  ScAddr solutionNode = ms_context->CreateNode(ScType::NodeConst);
-  // TODO: Add params to solution node
+  ScAddr const & solutionNode = ms_context->CreateNode(ScType::NodeConst);
   GenerationUtils::generateRelationBetween(ms_context, solutionNode, formula, CoreKeynodes::rrel_1);
+  ScAddr const & replacementsNode = ms_context->CreateNode(ScType::NodeConst);
+  GenerationUtils::generateRelationBetween(ms_context, solutionNode, replacementsNode, CoreKeynodes::rrel_2);
+  for (const auto & varName: varNames)
```

Comment:
Do not use auto for primitives (also strings)
```suggestion
  for (std::string const & varName : varNames)
```